src/fmeval/transforms/summarization_accuracy_metrics.py

```diff
@@ -36,67 +37,86 @@ class SummarizationAccuracyMetric(Transform):
+            `target_output_keys_provider`.
+        :param target_output_keys_provider: The key corresponding to a list of target
+            outputs. Will only be used if `target_output_keys` is not provided. This key
+            must not be "" otherwise it will be considered invalid.
         :param *args: Variable length argument list.
         :param **kwargs: Arbitrary keyword arguments.
         """
         assert_condition(
-            len(target_output_keys) == len(model_output_keys) and len(target_output_keys) == len(output_keys),
-            "len(target_output_keys), len(model_output_keys) and len(output_keys) should all match. "
-            f"len(target_output_keys) is {len(target_output_keys)}, len(model_output_keys) is "
+            len(model_output_keys) == len(output_keys),
+            "len(model_output_keys) and len(output_keys) should match. "
+            f"len(model_output_keys) is "
             f"{len(model_output_keys)}, and len(output_keys) is {len(output_keys)}.",
         )
+        assert_condition(
+            target_output_keys is not None or target_output_keys_provider != "",
+            f"target_output_keys is {target_output_keys}, and target_output_keys_provider"
+            f" is {target_output_keys_provider}."
+            "At least one must be valid.",
+        )
         super().__init__(
-            target_output_keys,
-            model_output_keys,
             output_keys,
+            model_output_keys,
             allow_duplicate_input_keys,
+            target_output_keys,
+            target_output_keys_provider,
             *args,
             **kwargs,
         )
+        input_keys = [target_output_keys_provider] if target_output_keys is None else target_output_keys
         self.register_input_output_keys(
-            target_output_keys + model_output_keys,
+            input_keys + model_output_keys,
             output_keys,
             allow_duplicates=allow_duplicate_input_keys,
         )
         self.target_output_keys = target_output_keys
         self.model_output_keys = model_output_keys
+        self.target_output_keys_provider = target_output_keys_provider
 
     @validate_call
     def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
         """Augment the input record with metrics computed via self.compute_metric.
+        The max score is computed over all possible targets represented by
+        self.target_output_keys and stored in the input record.
 
         :param record: The input record.
         :returns: The input record with metrics added in.
         """
-        for target_output_key, model_output_key, output_key in zip(
-            self.target_output_keys, self.model_output_keys, self.output_keys
-        ):
-            score = self.compute_metric(record[target_output_key], record[model_output_key])
-            record[output_key] = score
+        if self.target_output_keys is not None:
```

Comment:
You can reuse the for loop by setting `target_output_list = self.target_output_keys if self.target_output_keys else record[self.target_output_keys_provider]`